RecoTracker/FinalTrackSelectors/plugins/TrackCollectionFilterCloner.cc

```diff
@@ -0,0 +1,171 @@
+  TrackCollectionCloner::fill(clonerDesc);
+  desc.addUntracked<edm::ParameterSetDescription>("cloner",clonerDesc);
+  descriptions.add("TrackCollectionFilterCloner", desc);  
+}
+
+TrackCollectionFilterCloner::TrackCollectionFilterCloner(const edm::ParameterSet& iConfig) :
+  collectionCloner(*this, iConfig, true)
+  , originalTrackSource_  (iConfig.getParameter<edm::InputTag>("originalSource"),consumesCollector())
+  , originalMVAValsToken_  (consumes<MVACollection>        (iConfig.getParameter<edm::InputTag>("originalMVAVals") ) )
+  , originalQualValsToken_ (consumes<QualityMaskCollection>(iConfig.getParameter<edm::InputTag>("originalQualVals")) )
+  , minQuality_           (reco::TrackBase::qualityByName(iConfig.getParameter<std::string>("minQuality")) )
+{    
+  
+  produces<MVACollection>("MVAValues");
+  produces<QualityMaskCollection>("QualityMasks");
+
+
+}
+
+TrackCollectionFilterCloner::~TrackCollectionFilterCloner()
+{
+}
+
+void TrackCollectionFilterCloner::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const
+{
+
+  TrackCollectionCloner::Producer producer(iEvent, collectionCloner);
+ 
+  // load original tracks
+  auto const & originalsTracks = originalTrackSource_.tracks(iEvent);
+  //  auto const & originalIndices = originalTrackSource_.indicesInput(iEvent);
+  auto nTracks = originalsTracks.size();
+
+  edm::Handle<MVACollection> originalMVAStore;
+  iEvent.getByToken(originalMVAValsToken_,originalMVAStore);
+  assert((*originalMVAStore).size()==nTracks);
+
+  edm::Handle<QualityMaskCollection> originalQualStore;
+  iEvent.getByToken(originalQualValsToken_,originalQualStore);
+  assert((*originalQualStore).size()==nTracks);
+
+  // define minimum quality as set in the config file
+  unsigned char qualMask = ~0;
+  if (minQuality_!=reco::TrackBase::undefQuality) qualMask = 1<<minQuality_; 
+
+  // products
+  std::vector<unsigned int> selId;
+  auto pmvas  = std::make_unique<MVACollection>();
+  auto pquals = std::make_unique<QualityMaskCollection>();                     
+
+  auto k=0U;
+  for (auto j=0U; j<nTracks; ++j) {
+    if (! (qualMask&(* originalQualStore)[j]) ) continue;
+
+    selId.push_back(j);
+    pmvas->push_back((*originalMVAStore) [j]);
+    pquals->push_back((*originalQualStore) [j]);
+
+    ++k;
+  }
+
+  // clone selected tracks...
+  auto nsel = k;
+  auto isel = 0U;
+  assert(producer.selTracks_->size()==isel);
+  producer(originalTrackSource_,selId);
+  assert(producer.selTracks_->size()==nsel);
+
+  for (;isel<nsel;++isel) {
+    auto & otk = (*producer.selTracks_)[isel];
+    otk.setQualityMask( (*pquals)[isel] );
+  }  
+  assert(producer.selTracks_->size()==pmvas->size() );
+  assert(producer.selTracks_->size()==pquals->size());
+  
+  iEvent.put(std::move(pmvas),"MVAValues");
+  iEvent.put(std::move(pquals),"QualityMasks");
+
+
+}
+
+}
+
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
```

Comment:
@VinInn: OK, I see your point, particularly since the include statements go with the DEFINE_FWK_MODULE line.